bin/bump-version.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF
+Script for bumping version number.
+
+It provides functionality for automatically:
+ • Creating GitHub pull requests for bumping version numbers
+
+USAGE
+    bash bin/bump-version.sh [-v|--verbose] [-n|--dry-run]
+
+OPTIONS
+    -v | --verbose      Verbose output
+    -n | --dry-run      Dry run; don't create any commits.
+
+EXAMPLES
+    bash bin/bump-version.sh
+EOF
+}
+
+verbose="n"
+dryrun="n"
+while [[ $# -gt 0 ]]; do
+	case "$1" in
+	-v|--verbose)
+		verbose="y"
+		;;
+	-n|--dry-run)
+		dryrun="y"
+		;;
+	*)
+		usage
+		exit 1
+		;;
+	esac
+	shift
+done
+
+log() {
+	if [[ "$verbose" == "y" ]]; then
+		echo "$@"
+	fi
+}
+
+if [[ -n "$(git status --porcelain)" ]]; then
+	echo "ERROR: Dirty Git index, please commit all changes before continuing" 1>&2
+	if [[ "$dryrun" == "n" ]]; then
+		exit 1
+	fi
+fi
+if ! command -v gh &> /dev/null; then
+	echo "ERROR: Please install the 'gh' GitHub CLI" 1>&2
+	exit 1
+fi
+
+name="$(jq -r '.name' package.json)"
+current="$(jq -r '.version' package.json)"
+latest="$(npm view "$name" version)"
+
+log "current: v$current"
+log "latest:  v$latest"
+
+if ! printf "%s\n" "$latest" "$current" | sort -C -V; then
+	echo "ERROR: Latest NPM version is newer than current version" 1>&2
+	exit 1
+fi
+
+tag="v$current"
+commit="$(git rev-parse HEAD)"
+draft="$(gh release view "$tag" --json isDraft,targetCommitish --jq 'if(.isDraft) then .targetCommitish else "" end' 2>/dev/null || true)"
+
+log "commit:  ${commit}"
+log "draft:   ${draft:-none}"
+
+if [[ -z "$draft" ]] && [[ "$current" == "$latest" ]]; then
```

Comment:
There is so much duplicate code here...

Can't we just change the `github-release.sh` script to have verbs and then create separate GitHub actions with each verb?

For example:

```
bash bin/github-release.sh bump
```